nrf_rpc/nrf_rpc.c

```diff
@@ -1194,6 +1234,46 @@ int nrf_rpc_init(nrf_rpc_err_handler_t err_handler)
 	return err;
 }
 
+void nrf_rpc_register_callback_cleanup(struct callback_cleanup_handler * handler)
+{
+	k_mutex_lock(&cleanup_mutex, K_FOREVER);
+
+	struct callback_cleanup_handler * current;
+
+	// make sure not added twice
+	for (current = cleanup_handlers; current != NULL; current = current->next) {
+		if (current == handler) {
+			break;
+		}
+	}
+
+	if (current == NULL) {
+		handler->next = cleanup_handlers;
+		cleanup_handlers = handler;
+	}
+	k_mutex_unlock(&cleanup_mutex);
+}
+
+void nrf_rpc_stop(bool unregister_cb)
+{
+	k_mutex_lock(&cleanup_mutex, K_FOREVER);
+
+	is_stopped = true;
+	abort_all_ops();
+	if (unregister_cb) {
```

Comment:
Is there a reason why this needs to be done inside `nrf_rpc_stop`? My "concern" is that callback is not really a concept of nRF RPC but of the upper layer and this code is mixing two layers. I thought that after `nrf_rpc_stop` no further packets should be processed so it should be safe to clean callbacks in the caller.